### ISSUE TITLE
FreeBSD 11 agent-0.47 py2.7 - phpfpm_meta failed to find php-fpm bin path

### ISSUE THREAD

**Issue by actionm** [nginxinc/nginx-amplify-agent#79]
FreeBSD 11.1-RELEASE-p1

```
2017-11-06 13:19:57,068 [47284] phpfpm_meta failed to find php-fpm bin path, last attempt: "ls -la /proc/47263/exe" failed due to AmplifySubprocessError
2017-11-06 13:19:57,069 [47284] phpfpm_meta additional info:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/nginx_amplify_agent-0.47-py2.7.egg/amplify/ext/phpfpm/collectors/master/meta.py", line 76, in bin_path
    ls, _ = subp.call(ls_cmd)
  File "/usr/local/lib/python2.7/site-packages/nginx_amplify_agent-0.47-py2.7.egg/amplify/agent/common/util/subp.py", line 32, in call
    raise AmplifySubprocessError(message=command, payload=dict(returncode=process.returncode))
AmplifySubprocessError: (message=ls -la /proc/47263/exe, payload={'returncode': 1})


root@server:/root# ls -la /proc/47263/
total 0
dr-xr-xr-x  1 user  user   0 Nov  6 13:21 ./
dr-xr-xr-x  1 root  wheel  0 Nov  6 13:21 ../
-r--r--r--  1 user  user   0 Nov  6 13:21 cmdline
----------  1 user  user   0 Nov  6 13:21 ctl
----------  1 user  user   0 Nov  6 13:21 dbregs
-r--r--r--  1 user  user   0 Nov  6 13:21 etype
lr--r--r--  1 user  user   0 Nov  6 13:21 file@ -> /usr/local/sbin/php-fpm
----------  1 user  user   0 Nov  6 13:21 fpregs
-r--r--r--  1 user  user   0 Nov  6 13:21 map
----------  1 user  user   0 Nov  6 13:21 mem
----------  1 user  user   0 Nov  6 13:21 note
----------  1 user  user   0 Nov  6 13:21 notepg
----------  1 user  user   0 Nov  6 13:21 osrel
----------  1 user  user   0 Nov  6 13:21 regs
-r--r--r--  1 user  user   0 Nov  6 13:21 rlimit
-r--r--r--  1 user  user   0 Nov  6 13:21 status
```

**Comment by RvdE** [comment]
Try this patch. Works for me.

[amplify_ext_phpfpm_util_ps.py.patch.txt](https://github.com/nginxinc/nginx-amplify-agent/files/1621651/amplify_ext_phpfpm_util_ps.py.patch.txt)

**Comment by gshulegaard** [comment]
Thanks for the input @RvdE!  Unfortunately I don't believe this will help this particular case since the issue being reported is that there is no `exe` interface in the `/proc/<pid>/` tree.  Your patch appears to alter our `ps` parsing behavior which we do before `ls` in order to gather the appropriate `pid` to use.

**Comment by RvdE** [comment]
Yes. But this is how my ps output looks on FreeBSD 11.1:
 1012  -  Ss        0:15.08 php-fpm: master process (/usr/local/etc/php-fpm.conf) (php-fpm)
 3278  -  I         0:04.69 php-fpm: pool www (php-fpm)
37931  -  I         0:02.91 php-fpm: pool www (php-fpm)
76147  -  S         0:02.19 php-fpm: pool www (php-fpm)

As you can see there's additional info at the end of the lines, which is why I modify the PS regex. Otherwise the regex doesn't return any PID.

**Comment by gshulegaard** [comment]
Different `ps` outputs is another known issue and I understood why you made the change; I was just advising that it would not help for _this_  reported issue.

**Comment by gshulegaard** [comment]
Different `ps` outputs is another known issue and I understood why you made the change; I was just advising that it would not help for _this_  reported issue.

**Comment by dedm** [comment]
this one should be fixed in 1.1.0 (commit https://github.com/nginxinc/nginx-amplify-agent/commit/ad297e1155e94ddffbc8be2fd9397ac200936e0f)
